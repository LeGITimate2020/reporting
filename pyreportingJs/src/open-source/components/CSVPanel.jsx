import React from 'react';
import { uniqueId } from 'lodash';
import CSVFileParser from './CSVFileParser';
import TableExplorer from './TableExplorer';

/**
 * Higher level component to combine CSVFileParser to TableExplorer
 * - passes data and filename to Table Explorer
 */
export default class CSVPanel extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      fileName: null,
      data: [],
      fileId: 0,
    };
  }

  /**
   * @param values {Object} -- Papa.parsed object
   * @param fileName {String}
   */
  onCSVImport = (values, fileName) => {
    this.setState({
      fileName,
      data: values.data,
      fileId: uniqueId('CSVPanel'),
    });
  };

  render() {
    return (
      <div className="row">
        <div className="col-lg-6">
          <CSVFileParser onLoad={this.onCSVImport} />
        </div>
        <div className="col-lg-12">
          <TableExplorer key={this.state.fileId} data={this.state.data} title={this.state.fileName || 'CSV Content'} />
        </div>
      </div>
    );
  }
}
