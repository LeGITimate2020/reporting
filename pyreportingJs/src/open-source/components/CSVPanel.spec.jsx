import { shallow } from 'enzyme';
import React from 'react';
import CSVPanel from './CSVPanel';

describe('CSVPanel.jsx', () => {
  const csvData = [
    ['header1', 'header2', 'header3'],
    [11, 21, 31],
    [12, 22, 32],
    [13, 23, 33],
  ];

  it('renders without crashing', () => {
    shallow(<CSVPanel />);
  });

  it('Loads CSV data into component state', () => {
    const wrapper = shallow(<CSVPanel />);

    // Run PCA button should not exist
    expect(wrapper.find('.btn-run-pca').length).toBe(0);

    // Tab strip should not exist
    expect(wrapper.find('.csv-panel-tab').length).toBe(0);

    // no data should be loaded into state
    expect(wrapper.instance().state.data.length).toBe(0);

    // Simulate dropping CSV file
    wrapper.instance().onCSVImport({ data: csvData }, 'mySampleFile.csv');

    // data should be in state now
    expect(wrapper.instance().state.data.length).toBe(csvData.length);
    expect(wrapper.instance().state.data).toEqual(csvData);
  });

  it('Matches snapshot', () => {
    const wrapper = shallow(<CSVPanel />);
    expect(wrapper).toMatchSnapshot();
  });
});
