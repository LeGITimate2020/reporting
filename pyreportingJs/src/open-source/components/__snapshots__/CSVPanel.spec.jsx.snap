// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CSVPanel.jsx Matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CSVPanel />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="col-lg-6"
        >
          <CSVFileParser
            onLoad={[Function]}
          />
        </div>,
        <div
          className="col-lg-12"
        >
          <TableExplorer
            data={Array []}
            shouldRunPCA={false}
            title="CSV Content"
          />
        </div>,
      ],
      "className": "row",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <CSVFileParser
            onLoad={[Function]}
          />,
          "className": "col-lg-6",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onLoad": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <TableExplorer
            data={Array []}
            shouldRunPCA={false}
            title="CSV Content"
          />,
          "className": "col-lg-12",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "data": Array [],
            "shouldRunPCA": false,
            "title": "CSV Content",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col-lg-6"
          >
            <CSVFileParser
              onLoad={[Function]}
            />
          </div>,
          <div
            className="col-lg-12"
          >
            <TableExplorer
              data={Array []}
              shouldRunPCA={false}
              title="CSV Content"
            />
          </div>,
        ],
        "className": "row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <CSVFileParser
              onLoad={[Function]}
            />,
            "className": "col-lg-6",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onLoad": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <TableExplorer
              data={Array []}
              shouldRunPCA={false}
              title="CSV Content"
            />,
            "className": "col-lg-12",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "data": Array [],
              "shouldRunPCA": false,
              "title": "CSV Content",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
